src/openshift/project.ts

```diff
@@ -19,25 +19,42 @@ export class Project extends OpenShiftItem {
     static async set(): Promise<string | null> {
         let message: string = null;
         const kind = await getNamespaceKind();
+        const canCreateProjects = await Oc.Instance.canCreateNamespace();
+        const canListProjects = await Oc.Instance.canListNamespaces();
+
         const createNewProject = {
             label: `Create new ${kind}`,
             description: `Create new ${kind} and make it active`
         };
+        const manuallySetProject = {
+            label: `Manually set active ${kind}`,
+            description: `Type in ${kind} name and make it active`
+        };
         const projectsAndCreateNew = Oc.Instance
             .getProjects() //
-            .then((projects) => [
-                createNewProject,
-                ...projects.map((project) => ({
-                    label: project.name,
-                    description: project.active ? 'Currently active': '',
-                })),
-            ]);
+            .then((projects) => {
+                const items = [];
+                if (canCreateProjects) {
+                    items.push(createNewProject);
+                }
+                items.push(manuallySetProject);
+                if (canListProjects) {
+                    items.push(...projects.map((project) => ({
+                        label: project.name,
+                        description: project.active ? 'Currently active': '',
+                    })));
+                }
+                return items;
+            });
         const selectedItem = await window.showQuickPick(projectsAndCreateNew, {placeHolder: `Select ${kind} to activate or create new one`});
         if (!selectedItem) return null;
         if (selectedItem === createNewProject) {
             await commands.executeCommand('openshift.project.create');
         } else {
-            const projectName = selectedItem.label;
+            const projectName = selectedItem === manuallySetProject ?
+                await Project.getProjectName(`${kind} name`, new Promise((resolve) => {resolve([])})) : selectedItem.label;
```

Comment:
For the `new Promise((resolve) => { resolve([]) })` you can instead use `Promise.resolved([])`

Reply:
Fixed